### ISSUE TITLE
Fails on "record" variable invocation

### ISSUE THREAD

**Issue by famod** [javaparser/javaparser#3261]
## Short story

This parser fails on:
```java
Object record = new Object();
record.toString();
```

## Long story

Reproducer: https://github.com/famod/java-parser-record-var

https://github.com/revelc/impsort-maven-plugin which is using this parser fails with:
```
(line 6,col 16) 'record' cannot be used as an identifier as it is a keyword.
(line 7,col 9) Parse error. Found ".", expected one of  "enum" "exports" "module" "open" "opens" "provides" "record" "requires" "strictfp" "to" "transitive" "uses" "with" "yield" <IDENTIFIER>
```
So it fails here: https://github.com/famod/java-parser-record-var/blob/main/src/main/java/javaparser/RecordVar.java#L7

AFAICS, `record` is **not** a generally reserved keyword like e.g. `final`, it is only restriced in some contexts.
Given that the code compiles just fine on JDK 16 underlines this observation.

See also: https://github.com/revelc/impsort-maven-plugin/issues/44#issuecomment-840766589

**Comment by ctubbsii** [comment]
Here's a minimal test case for JUnit that demonstrates the problem:

```java
import static org.junit.Assert.assertTrue;

import java.io.FileNotFoundException;

import org.junit.jupiter.api.Test;

import com.github.javaparser.JavaParser;
import com.github.javaparser.ParserConfiguration;
import com.github.javaparser.ParserConfiguration.LanguageLevel;
import com.github.javaparser.Problem;

class RecordVarParseTest {

  @Test
  void testRecordAsIdentifier() throws FileNotFoundException {
    String fileContents = """
        public class RecordVar {
          public void foo() {
            Object record = new Object();
            record.toString();
          }
        }
        """;
    var parseConfig = new ParserConfiguration().setLanguageLevel(LanguageLevel.JAVA_16);
    var parseResult = new JavaParser(parseConfig).parse(fileContents);
    var problems = parseResult.getProblems();
    problems.stream().map(Problem::getMessage).forEach(System.out::println);
    assertTrue(problems.isEmpty()); // this fails with 3.22.0
  }

}
```

**Comment by ctubbsii** [comment]
This test case fails, even if the parser is configured with `LanguageLevel.JAVA_11`

**Comment by zhangziqiang1** [comment]
This problem has been resolved, and you can look at #3255 [https://github.com/javaparser/javaparser/issues/3255](https://github.com/javaparser/javaparser/issues/3255)

**Comment by famod** [comment]
@zhangziqiang1 thanks! Fix confirmed with curent `master`.
